Don’t auto-select a pullquote text colour for regular style

johngodley opened the pull request:
The pullquote block auto-selects an appropriate text colour based on the main colour. This is designed to give a good contrast. However it assumes the quote has a 'solid colour' style where the main colour is the background, with the text overlayed:

![edit_post_ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/47211137-a6ac6c00-d38c-11e8-98d8-df2183311608.jpg)

If the pullquote uses the 'regular colour' style the background is white (or whatever the theme uses), and the main colour changes the top and bottom border:

![edit_post_ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/47211237-e7a48080-d38c-11e8-87b7-d0b6c81156cb.jpg)

The colour calculated for the main colour background can often give a poor contrast ratio on this white background.

For example, if you create a pull quote with regular style and pick red as the main colour it sets the text colour to light grey (the calculated text colour for red). In the above solid colour quote this gives good contrast, but on a regular colour quote with white background it gives poor contrast:

![edit_post_ _wordpress_latest_ _wordpress](https://user-images.githubusercontent.com/1277682/47211340-1e7a9680-d38d-11e8-92bb-333723afc407.jpg)

This PR restricts the auto-colour selection to the 'solid colour' style only. I'm not sure if this fits the intended purpose of the block, but it fixes the problem in this instance.

Also fixes the situation where clicking 'clear' on the main colour sets a text colour.

Fixes #10568

## How has this been tested?
1. Create a pull quote block
2. Open block settings
3. Set main colour to red
4. Confirm the text colour does not change
5. Clear the text colour
6. Clear the main colour and verify that the text colour is not set

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

johngodley committed:
Don’t auto-select a pullquote text colour for regular style

Regular style uses the theme background, but the auto-selected colour assumes the main colour is the background. This can often lead to a very poor colour choice.

johngodley committed:
Only auto-set colour if setting the main colour

This prevents clearing of the main colour from setting the text colour

jorgefilipecosta commented on packages/block-library/src/pullquote/edit.js:
```diff
@@ -33,9 +33,12 @@ class PullQuoteEdit extends Component {
+
 		setMainColor( colorValue );
-		if ( ! textColor.color || this.wasTextColorAutomaticallyComputed ) {
+		if ( shouldSetTextColor || this.wasTextColorAutomaticallyComputed ) {
```
If we go the solid color style we pick the main color, and then we change to the regular style, each time we change the main color the text color continue to change.
I think maybe we can solve this by updating shouldSetTextColor to be `isSolidColorStyle && colorValue && ( ! textColor.color || this.wasTextColorAutomaticallyComputed );`

johngodley replied:
Ah yes, nice! Updated

jorgefilipecosta reviewed (commented):
Thank you for this change @johngodley this generally works well on my tests 👍  There is only an edge case that I think we may be able to improve.

johngodley committed:
Stop switch from solid to regular style still changing text colour

If you picked the solid style and set a main colour and then switched to regular style and picked a main colour it would continue to change the text colour. This moves the wasTextColorAutomaticallyComputed check so this doesn’t happen

jorgefilipecosta reviewed (approved):
LGTM 👍
